src/core/overlay.ts

```diff
@@ -0,0 +1,96 @@
+import * as Config from '@oclif/config';
+import debug from 'debug';
+import jsonpath from 'jsonpath';
+import mergician from 'mergician';
+import { JSONSchema4Object } from 'json-schema';
+import { safeStringify } from '@stoplight/yaml';
+
+import { APIDefinition, OpenAPIOverlay } from '../definition';
+
+export class Overlay {
+  _config: Config.IConfig;
+
+  public constructor(config: Config.IConfig) {
+    this._config = config;
+  }
+
+  public async run(
+    definition: APIDefinition,
+    overlay: OpenAPIOverlay,
+    format: string,
+  ): Promise<string | undefined> {
+    const newDefinition = this.applyOverlay(definition, overlay);
+
+    if (format == 'json') {
+      return JSON.stringify(newDefinition);
+    } else {
+      return safeStringify(newDefinition);
+    }
+  }
+
+  // WIP @github.com/lornajane/openapi-overlays-js
+  //
+  // I couldn't get the upstream lib to be imported properly due to
+  // some issues with ESM module imports so this is method was copied
+  // from github.com/lornajane/openapi-overlays-js and has been
+  // adapted to make our Typescript build happy.
+  //
+  // If you make any changes here, PLEASE ALSO MAKE THEM UPSTREAM.
+  applyOverlay(spec: APIDefinition, overlay: OpenAPIOverlay): APIDefinition {
```

Comment:
It feel like this method could be a bit enhanced by extracting smaller methods, having a case to split the different actions, etc...

It can be done later, but I think it would be a great idea to make it a bit clearer. WDYT?

Reply:
The point here was to stay as close as possible from the original code @github.com/lornajane/openapi-overlays-js. The code will soon be available as a library and we will be able to build a common ground in a dedicated package which will remove this code from our CLI.

I'd prefer to keep it as is for now, if you don't mind?